framework/source/class/qx/ui/form/AbstractField.js

```diff
@@ -213,7 +213,17 @@ qx.Class.define("qx.ui.form.AbstractField",
       check : "Boolean",
       init : false
     },
-
+    /**
+     * Fire a {@link #changeValue} event whenever the content of the
+     * field matches the given regular expression. Accepts both regular
+     * expression objects as well as strings for input.
+     */
+    liveUpdateOnRxMatch :
+    {
+      check : "RegExp",
```

Comment:
If the property setter checks for "RegExp" and throws an error if the value isn't one, you wouldn't be able to store a string in the first place, would you? I think to be able to set both is good, since the value could be bound from a JSON-marshalled object.

Reply:
the transform is applied before the check

Reply:
I see!

Reply:
You don't actually need a "check" and it's confusing to have it as "RegExp" when you're really expecting a string as input. Instead, the transform function can throw an error to also indicate that validation fails. That would be a cleaner way of doing this, I think.

Reply:
Well that is the way we describe the use case for the transform function in the documentation ... I am not expecting a string as input ... you can use both, a string OR a regular expression.

Reply:
My point is that it looks like magic, as is. If `check` says RegExp then that's what will show up in the api docs, and the user will only think it can be that type. Maybe change it to:

```
check : function(value)
{
  return value instanceof RegExp || typeof value == "string";
}
```
Granted, by the time the function is called, it will always be a RegExp if what was passed in was a string, but this shows what the range of inputs can be, and provided the required checking as well.

If you feel strongly about leaving it as is, I'll withdraw my objection, but I do have a preference for something that accurately documents the legal range of inputs.

Reply:
The issue here really is about the auto-generated documentation ... that is why I documented the special behavior explicitly in the text. And yes I do feel strongly about it since this is how the transform property is supposed to be used.

Maybe we can think of a way to fix the documentation maybe integrating some jsdoc supplied for the transform function?